lib/AST/ASTContext.cpp

```diff
@@ -2912,6 +2912,18 @@ Type ErrorType::get(Type originalType) {
   return entry = new (mem) ErrorType(ctx, originalType, properties);
 }
 
+Type UnresolvedType::get(Type originalType) {
+  assert(originalType);
+
+  auto originalProperties = originalType->getRecursiveProperties();
+  auto arena = getArena(originalProperties);
+
+  auto &ctx = originalType->getASTContext();
+  void *mem = ctx.Allocate(sizeof(UnresolvedType) + sizeof(Type),
+                           alignof(UnresolvedType), arena);
+  return new (mem) UnresolvedType(nullptr, originalType);
```

Comment:
Please change this to only allocate one unresolved type per original type. We want all types to be uniqued